### PR TITLE
Vadym.hutovskyi

### PR DESCRIPTION
**Pull request by Vadym-tst**
Corrected according to last remarks.

### CODE REVIEW

**Comment by alexposukhov** [comment]
Please set label "Ready for review" if your change is ready

File: 01_git/scissors/main.c [hunk 1]
```diff
@@ -12,17 +12,18 @@ int main()
         SCISSORS,
         VARS_NUM
     };
-    char input[16];
-    int player, ai;
-    const char* option[] = {"rock", "paper", "scissors"};
+    const uint8_t buf_lenght = 16;
```
Comment (ziod) [inline 01_git/scissors/main.c:15]: static or #define ... general C approach

File: 01_git/scissors/main.c [hunk 2]
```diff
@@ -40,18 +40,23 @@ int main()
...
+                printf("You win: %s beats %s\n\n", option[player], option[ai]);
+            } else {
+                printf("I win: %s beats %s\n\n", option[ai], option[player]);
+            }
         }
-        game_active = 0;
+        
+        printf("Do you want to play again? Type (y) if YES or any key if NOT and press Enter\n");
+        fgets(buf, sizeof(buf), stdin);
+        if(buf[0]!='y')
```
Comment (ziod) [inline 01_git/scissors/main.c:57]: 'y' is a magic number all constants shall be defined to increase flexibility

File: 02_bash/hwdetect/hwdetect.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+while [ 1 ]
+do
+
+    echoDiff storage_devs.txt sd
+    echoDiff tty_devs.txt tty[UA]
+    echoDiff i2c_devs.txt i2c
+    echoDiff sd_cards.txt mmcblk
+
+done
```
Comment (ziod) [inline 02_bash/hwdetect/hwdetect.sh]: no new line at the end

File: 02_bash/hwdetect/hwdetect.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+function echoDetected() {
+
```
Comment (ziod) [inline 02_bash/hwdetect/hwdetect.sh:4]: please remove spaces 4, 15, 18 rows. It is appliable for all 3 files in PR.

File: 02_bash/hwdetect/hwdetect.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+﻿#!/bin/bash
```
Comment (ziod) [inline 02_bash/hwdetect/hwdetect.sh:1]: please squash with the previous.

**Review by ziod — CHANGES_REQUESTED** [review]

File: 03_module/TASK3/TASK3.c [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (ziod) [inline 03_module/TASK3/TASK3.c:1]: WARNING: Missing or malformed SPDX-License-Identifier tag in line 1
#1: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:1:
+

ERROR: trailing whitespace
#5: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:5:
+#include <linux/kernel.h> $

ERROR: trailing whitespace
#6: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:6:
+ $

WARNING: please, no spaces at the start of a line
#6: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:6:
+ $

ERROR: space required after that ',' (ctx:VxV)
#8: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:8:
+module_param(param_1,int,0660);
                     ^

ERROR: space required after that ',' (ctx:VxV)
#8: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:8:
+module_param(param_1,int,0660);
                         ^

ERROR: trailing whitespace
#9: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:9:
+ $

WARNING: please, no spaces at the start of a line
#9: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:9:
+ $

ERROR: space required after that ',' (ctx:VxV)
#11: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:11:
+module_param(param_2,int,0660);
                     ^

ERROR: space required after that ',' (ctx:VxV)
#11: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:11:
+module_param(param_2,int,0660);
                         ^

ERROR: trailing whitespace
#12: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:12:
+ $

WARNING: please, no spaces at the start of a line
#12: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:12:
+ $

WARNING: Prefer [subsystem eg: netdev]_warn([subsystem]dev, ... then dev_warn(dev, ... then pr_warn(...  to printk(KERN_WARNING ...
#15: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:15:
+	printk(KERN_WARNING "Start: param_1=%d, param_2=%d, param_1+param_2=%d\n", param_1, param_2, param_1+param_2);

ERROR: trailing whitespace
#18: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:18:
+ $

WARNING: please, no spaces at the start of a line
#18: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:18:
+ $

WARNING: Prefer [subsystem eg: netdev]_warn([subsystem]dev, ... then dev_warn(dev, ... then pr_warn(...  to printk(KERN_WARNING ...
#21: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:21:
+	printk(KERN_WARNING "Stop: param_1=%d, param_2=%d, param_1-param_2=%d\n", param_1, param_2, param_1-param_2);

ERROR: trailing whitespace
#23: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:23:
+ $

WARNING: please, no spaces at the start of a line
#23: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/TASK3/TASK3.c:23:
+ $

total: 10 errors, 8 warnings, 30 lines checked

**Review by ziod — CHANGES_REQUESTED** [review]

File: 04_basic_struct/TASK4/TASK4.c [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	int count = 0;
+	struct s_list *element;
+
+	if (list_empty(&top_head)) {
+		count = sprintf(buf, "There is no records yet...\n");
+	}
+	list_for_each_entry(element, &top_head, node) {
+		count += sprintf(&buf[count], "%s", element->text);
+	}
+	return count;
```
Comment (ziod) [inline 04_basic_struct/TASK4/TASK4.c:33]: strlen(buf)

**Review by ziod — CHANGES_REQUESTED** [review]

**Pull request closed without merging**